src/main/java/it/pagopa/pn/deliverypush/action/startworkflowrecipient/StartWorkflowForRecipientHandler.java

```diff
@@ -35,11 +36,11 @@ public StartWorkflowForRecipientHandler(CourtesyMessageUtils courtesyMessageUtil
         this.auditLogBuilder = pnAuditLogBuilder;
     }
 
-    public void startNotificationWorkflowForRecipient(String iun, int recIndex) {
+    public void startNotificationWorkflowForRecipient(String iun, int recIndex, RecipientsWorkflowDetails details) {
         log.info("Start notification workflow for recipient - iun {} id {}", iun, recIndex);
         
         NotificationInt notification = notificationService.getNotificationByIun(iun);
-
+        notification.getRecipients().get(recIndex).toBuilder().quickAccessLinkToken(details.getQuickAccessLinkToken());
```

Comment:
Andrei qui a creare un metodo chiamato ad esempio "**unrichNotificationWithQuickAccessLinkToken**" o qualcosa di questo genere, in modo che da lettore ho una lettura più semplice e veloce di ciò che viene fatto, senza dover entrare nel dettaglio.
Eviterei inoltre di inserire il tutto in un unica riga di codice, in modo da individuare più semplicemente eventuali NullPointer ecc.